airflow/providers/hashicorp/common/vault_client.py

```diff
@@ -0,0 +1,389 @@
+from hvac.api.auth_methods import github, kubernetes, ldap, radius, userpass
+from hvac.api.secrets_engines import aws, azure, gcp, kv_v1, kv_v2
+from hvac.exceptions import InvalidPath, VaultError
+from requests import Response
+
+from airflow.utils.log.logging_mixin import LoggingMixin
+
+VALID_KV_VERSIONS: List[int] = [1, 2]
+VALID_AUTH_TYPES: List[str] = [
+    'approle',
+    'aws_iam',
+    'azure',
+    'github',
+    'gcp',
+    'kubernetes',
+    'ldap',
+    'radius',
+    'token',
+    'userpass'
+]
+
+
+class VaultClient(LoggingMixin):  # pylint: disable=too-many-instance-attributes
+    """
+    Retrieves Authenticated client from Hashicorp Vault
+
+    :param url: Base URL for the Vault instance being addressed.
+    :type url: str
+    :param auth_type: Authentication Type for Vault. Default is ``token``. Available values are in
+        :py:const:`airflow.providers.hashicorp.common.vault_client.VALID_AUTH_TYPES`.
+    :type auth_type: str
+    :param mount_point: The "path" the secret engine was mounted on. Default depends on the engine used.
+    :type mount_point: str
+    :param kv_engine_version: Select the version of the engine to run (``1`` or ``2``, default: ``2``)
+    :type kv_engine_version: int
+    :param token: Authentication token to include in requests sent to Vault.
+        (for ``token`` and ``github`` auth_type)
+    :type token: str
+    :param username: Username for Authentication (for ``ldap`` and ``userpass`` auth_types)
+    :type username: str
+    :param password: Password for Authentication (for ``ldap`` and ``userpass`` auth_types)
+    :type password: str
+    :param key_id: Key ID for Authentication (for ``aws_iam`` and ''azure`` auth_type)
+    :type  key_id: str
+    :param secret_id: Secret ID for Authentication (for ``approle``, ``aws_iam`` and ``azure`` auth_types)
+    :type secret_id: str
+    :param role_id: Role ID for Authentication (for ``approle``, ``aws_iam`` and ``kubernetes`` auth_types)
+    :type role_id: str
+    :param kubernetes_role: Role for Authentication (for ``kubernetes`` auth_type)
+    :type kubernetes_role: str
+    :param kubernetes_jwt_path: Path for kubernetes jwt token (for ``kubernetes`` auth_type, default:
+        ``/var/run/secrets/kubernetes.io/serviceaccount/token``)
+    :type kubernetes_jwt_path: str
+    :param gcp_key_path: Path to GCP Credential JSON file (for ``gcp`` auth_type).
+           Mutually exclusive with gcp_keyfile_dict
+    :type gcp_key_path: str
+    :param gcp_keyfile_dict: Dictionary of keyfile parameters. (for ``gcp`` auth_type).
+           Mutually exclusive with gcp_key_path
+    :type gcp_keyfile_dict: dict
+    :param gcp_scopes: Comma-separated string containing GCP scopes (for ``gcp`` auth_type)
+    :type gcp_scopes: str
+    :param azure_tenant_id: Tenant of azure (for ``azure`` auth_type)
+    :type azure_tenant_id: str
+    :param azure_resource: Resource if of azure (for ``azure`` auth_type)
```

Comment:
```suggestion
    :param azure_resource: Resource for azure (for ``azure`` auth_type)
```

or something like `The configured URL for the application registered in Azure Active Directory.`

Reply:
Done